core/src/main/java/com/sequenceiq/cloudbreak/service/stack/InstanceMetaDataService.java

```diff
@@ -55,14 +56,19 @@ public void updateInstanceStatus(final InstanceMetaData instanceMetaData, com.se
         }
     }
 
-    public Stack saveInstanceAndGetUpdatedStack(Stack stack, List<CloudInstance> cloudInstances, boolean save) {
+    public Stack saveInstanceAndGetUpdatedStack(Stack stack, List<CloudInstance> cloudInstances, boolean save, Set<String> hostNames) {
+        Iterator<String> hostNameIterator = hostNames.iterator();
         for (CloudInstance cloudInstance : cloudInstances) {
             InstanceGroup instanceGroup = getInstanceGroup(stack.getInstanceGroups(), cloudInstance.getTemplate().getGroupName());
             if (instanceGroup != null) {
                 InstanceMetaData instanceMetaData = new InstanceMetaData();
                 instanceMetaData.setPrivateId(cloudInstance.getTemplate().getPrivateId());
                 instanceMetaData.setInstanceStatus(com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.InstanceStatus.REQUESTED);
                 instanceMetaData.setInstanceGroup(instanceGroup);
+                if (hostNameIterator.hasNext()) {
+                    String hostName = hostNameIterator.next();
+                    instanceMetaData.setDiscoveryFQDN(hostName);
+                }
```

Comment:
it would be nice to have logs here to see when we set hostnames, at least for now